### PR TITLE
Es6 for good

### PR DESCRIPTION
**Pull request by theodesp**
-Code improvements
-Added babel-register
-Functionalise app

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -86,12 +120,18 @@ function getMiddleware (isDev) {
     date_format: 'DD-MM-YYYY'
   })
 
-  return isDev
-    ? morgan('dev')
-    : morgan('combined', {
-      stream: morganAccessLogStream,
-      skip: (req, res) => res.statusCode < 400
-    })
+  const morganConfig = do {
```
Comment (theodesp) [inline lib/index.js]: What out this baby here

Comment (stefanoschrs) [inline lib/index.js]: Why use it here? In this case that ternary isn't so bad

File: lib/index.js [hunk 2]
```diff
@@ -19,39 +20,74 @@ let logger
...
+// /**
+//  * Initialize logger
+//  * @param {Object} config - Logger configuration
+//  * @returns {Promise}
+//  */
 function init (config = {}) {
   return new Promise((resolve, reject) => {
-    myConfig = config
-    let streams = []
+    myConfig = Object.assign(config, CONFIG_DEFAULTS)
```
Comment (stefanoschrs) [inline lib/index.js]: Good one, totally forgot about the *assign*

File: lib/index.js (hunk 1 above)
Comment (theodesp) [inline lib/index.js]: It is more of a preference this one. Take a look at those examples here: http://babeljs.io/docs/plugins/transform-do-expressions/

File: lib/index.js (hunk 2 above)
Comment (stefanoschrs) [inline lib/index.js]: @theodesp P.S `Object.assign` had to be `Object.assign(CONFIG_DEFAULTS, config)` in order to overwrite the defaults with the user given

**Pull request merged**